### DISCUSSION TITLE
why should `Session.expire_on_commit` be set to `False` when using asyncio?

### DISCUSSION THREAD

**Discussion by 0-th** [sqlalchemy/sqlalchemy#11495] [Usage Questions] [answered]
I'd love to understand better, why `Session.expire_on_commit` should be set to `False` for async sessions.
I came across it in this part of the [docs](https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#preventing-implicit-io-when-using-asyncsession).

I also think I might have come across it somewhere else in the docs (not sure I can still find it) where it was recommended (probably for sync sessions) that `expire_on_commit` never be set to `False`.

TIA for your time.

**Comment by CaselIT** [comment, marked as answer]
Hi,

It's just a suggestion to avoid having to manually refresh an object after a commit. Using async you cannot lazy load expired attributes after a commit like you would in case of sync code.
Note that there is a way of having lazy load work with asyncio, see `AsyncAttrs` https://docs.sqlalchemy.org/en/20/orm/extensions/asyncio.html#sqlalchemy.ext.asyncio.AsyncAttrs

**Reply by Kokleng-Dev** [reply]
So, set it to False for async session, right?
